server/modules/record/manager/_recordManager/recordUpdateManager.js

```diff
@@ -93,28 +93,57 @@ const deleteRecordsPreview = async (surveyId) =>
  */
 const persistNode = async (user, survey, record, node,
                            nodesUpdateListener = null, nodesValidationListener = null, t = db) =>
-  await t.tx(async t =>
-    await _onNodesUpdate(
-      survey,
-      await NodeUpdateManager.persistNode(user, survey, record, node, t),
-      nodesUpdateListener,
-      nodesValidationListener,
-      t,
-    )
+  await t.tx(async t => {
+      await _beforeNodeUpdate(user, survey, record, node, t)
+
+      const nodesUpdated = await NodeUpdateManager.persistNode(user, survey, record, node, t)
+
+      const { record: updatedRecord, updatedNodesAndDependents } = await _onNodesUpdate(
+        survey,
+        nodesUpdated,
+        nodesUpdateListener,
+        nodesValidationListener,
+        t,
+      )
+
+      await _afterNodesUpdate(user, survey, updatedRecord, updatedNodesAndDependents, t)
+
+      return updatedRecord
+    }
   )
 
 const deleteNode = async (user, survey, record, nodeUuid,
                           nodesUpdateListener = null, nodesValidationListener = null, t = db) =>
-  await t.tx(async t =>
-    await _onNodesUpdate(
-      survey,
-      await NodeUpdateManager.deleteNode(user, survey, record, nodeUuid, t),
-      nodesUpdateListener,
-      nodesValidationListener,
-      t,
-    )
+  await t.tx(async t => {
+      await _beforeNodeUpdate(user, survey, record, node, t)
+
+      const nodesUpdated = await NodeUpdateManager.deleteNode(user, survey, record, nodeUuid, t)
+
+      const { record: updatedRecord, updatedNodesAndDependents } = await _onNodesUpdate(
+        survey,
+        nodesUpdated,
+        nodesUpdateListener,
+        nodesValidationListener,
+        t,
+      )
+      await _afterNodesUpdate(user, survey, updatedRecord, updatedNodesAndDependents, t)
+
+      return updatedRecord
```

Comment:
persistNode and deleteNode do the same thing, expect for the persist/delete node. should we have just a method with all the before and after update logic? e.g ```_updateNodeAndValidateKeys```?

Reply:
or something similiar, right

Reply:
why not to call the method for what it really does? the word perform doesn't really tell me what the method does.. don't you think so?